export/tasks_generator.go

```diff
@@ -236,32 +260,59 @@ type TasksGenerator struct {
 	successTasksGenerator SuccessTasksTreeGenerator
 }
 
+func GetStartTasks(repo, branch, fromCommitRef, toCommitRef, exportID string) ([]parade.TaskData, error) {
+	one, zero := 1, 0
+	data := StartData{
+		Repo:          repo,
+		Branch:        branch,
+		FromCommitRef: fromCommitRef,
+		ToCommitRef:   toCommitRef,
+		ExportID:      exportID,
+	}
+	body, err := json.Marshal(data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to serialize %+v: %w", data, err)
+	}
+
+	bodyStr := string(body)
+	idGen := taskIDGenerator(exportID)
+	tasks := make([]parade.TaskData, 1)
+	tasks[0] = parade.TaskData{
+		ID:                idGen.startedTaskID(),
+		Action:            StartAction,
+		Body:              &bodyStr,
+		StatusCode:        parade.TaskPending,
+		MaxTries:          &one,
+		TotalDependencies: &zero,
+	}
+	return tasks, nil
+}
+
 // NewTasksGenerator returns a generator that exports tasks from diffs to file operations under
 // dstPrefix.  It generates success files for files in directories matched by
 // "generateSuccessFor".
-func NewTasksGenerator(exportID string, dstPrefix string, generateSuccessFor func(path string) bool) *TasksGenerator {
+func NewTasksGenerator(exportID string, dstPrefix string, generateSuccessFor func(path string) bool, finishBody *string) *TasksGenerator {
 	const numTries = 5
 	dstPrefix = strings.TrimRight(dstPrefix, "/")
 	makeDestination := func(path string) string {
 		return fmt.Sprintf("%s/%s", dstPrefix, path)
 	}
 
 	return &TasksGenerator{
-		ExportID:           exportID,
-		DstPrefix:          dstPrefix,
-		GenerateSuccessFor: generateSuccessFor,
-		NumTries:           numTries,
-		makeDestination:    makeDestination,
-		idGen:              taskIDGenerator(exportID),
-		successTasksGenerator: NewSuccessTasksTreeGenerator(
-			exportID, generateSuccessFor, makeDestination),
+		ExportID:              exportID,
+		DstPrefix:             dstPrefix,
+		GenerateSuccessFor:    generateSuccessFor,
+		NumTries:              numTries,
+		makeDestination:       makeDestination,
+		idGen:                 taskIDGenerator(exportID),
+		successTasksGenerator: NewSuccessTasksTreeGenerator(exportID, generateSuccessFor, makeDestination, finishBody),
 	}
 }
 
 // Add translates diffs into many tasks and remembers "generate success" tasks for Finish.  It
 // returns some tasks that can already be added.
 func (e *TasksGenerator) Add(diffs catalog.Differences) ([]parade.TaskData, error) {
-	const initialSize = 1_000
+	const initialSize = 1000
```

Comment:
Why?  I _really_ like this form, it is much easier for me to read large numbers when their digits are grouped in 3s.

Reply:
GoLands linter threw an exception (it isn't updated to work with this form yet), which caused the auto formatter to stop working. so because it was a small number I allowed myself to change it.
I could change it back